cloud9_config.py

```diff
@@ -0,0 +1,123 @@
+CONFIG_FILE_NAME = 'roboMakerSettings.json'
+SIM_CONFIG_ID = 'sim-robot-with-simple-world'
+BUCKET_KEY = 'RoboMakerS3Bucket'
+CONFIG_KEYS = {
+    BUCKET_KEY: ['cfg.simulation.outputLocation', 'cfg.robotApp.s3Bucket', 'cfg.simulationApp.s3Bucket'],
+    'SimulationRole': ['cfg.simulation.iamRole'],
+    'DefaultSecurityGroupID': ['cfg.simulation.vpcConfig.securityGroups'],
+    'PublicSubnet1': ['cfg.simulation.vpcConfig.subnets'],
+    'PublicSubnet2': ['cfg.simulation.vpcConfig.subnets']
+}
+
+parser = argparse.ArgumentParser(description='Mars Rover RoboMaker Config')
+parser.add_argument('--cloudformation-stack-name', dest='cf_stack_name', type=str, help='Full name of the workshop seeding CloudFormation stack')
+parser.add_argument('--project-root-dir', type=str, default='aws-robomaker-sample-application-open-source-rover', 
+                    help="Path to the Mars Rover root directory, absolute or relative to the directory from which this script is executed")
+parser.add_argument('--config-target-dir', type=str, default='./', help="Path to a directory to save the configured '{}' file".format(CONFIG_FILE_NAME))
+args = parser.parse_args()
+
+print("Getting info from CloudFormation stack '{}'".format(args.cf_stack_name))
+cfn = boto3.client('cloudformation')
+
+stack_info = cfn.describe_stacks(StackName=args.cf_stack_name)
+outputs = stack_info['Stacks'][0]['Outputs']
+
+print('Configuring the RoboMaker menu')
+
+template_config_file = os.path.join(args.project_root_dir, CONFIG_FILE_NAME)
+print('- loading config template: {}'.format(template_config_file))
+with open(template_config_file) as fin:
+    config = json.loads(fin.read())
+
+sim_config = None    
+for c in config['runConfigurations']:
+    if c['id'] == SIM_CONFIG_ID:
+        sim_config = c
+        break
+if sim_config is None:
+    raise RuntimeError('{} not round in the template config'.format(SIM_CONFIG_ID))
+
+# We'll need bucket elsewhere too
+bucket = None
+for key in CONFIG_KEYS:
+    
+    # Find the value in the CFN outputs
+    val = None
+    for output in outputs:
+        if output['OutputKey'] == key:
+            val = output['OutputValue']
+            break
+    if val is None:
+        raise RuntimeError('{} not round in the CloudFormation output'.format(key))
+    if key == BUCKET_KEY:
+        bucket = val
+        
+    for target_keys in CONFIG_KEYS[key]:
+        cfg = sim_config
+        key_parts = target_keys.split('.')
+        for k in key_parts[:-1]:
+            cfg = cfg[k]
+        if isinstance(cfg[key_parts[-1]], list):
+            cfg[key_parts[-1]].append(val)
+        else:
+            cfg[key_parts[-1]] = val
+        
+out_config_file = os.path.join(args.config_target_dir, CONFIG_FILE_NAME)
+print('- saving config to {}'.format(out_config_file))
+with open(out_config_file, 'w') as fout:
+    fout.write(json.dumps(config, indent=2))
+ 
+launch_file = os.path.join(args.project_root_dir, LAUNCH_FILE_PATH)
```

Comment:
I like how this script populates the roboMakerSettings.json file. However, instead of modifying the launch file directly with parameters, you should instead simply set environment variables in the roboMakerSettings.json file, then in the launch file use optenv: 
`<arg name="model" default="$(optenv TURTLEBOT3_MODEL waffle_pi)" doc="model type [burger, waffle, waffle_pi]"/>`

In other labs, we have created Lambda function mocks that will pull cloudformation -- however, I do like how your script will more directly integrate with the Cloud9 menu bar.

https://github.com/aws-samples/multi-robot-fleet-sample-application/tree/master/setup/fleetLauncherApp

Reply:
yes, makes sense, I didn't like modifying launch file either but didn't see an alternative, env vars seem a right approach. Now need to learn how to set the env vars in  roboMakerSettings.json file